### ISSUE TITLE
RepoToolset overrides DeployExtension set in the project file

### ISSUE THREAD

**Issue by sharwell** [dotnet/roslyn-tools#153]
`DeployExtension` is a standard VSSDK property that controls extension deployment. Here is an example usage:
https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/d18b15f2342b52800c86a9540f95c1a51b0ddbbb/StyleCop.Analyzers/StyleCop.Analyzers.Vsix/StyleCop.Analyzers.Vsix.csproj#L21-L24

This approach does not work with RepoToolset because the `DeployExtension` property is overwritten with the value from `DeployProjectOutput`. This is unintuitive and results in example VS extension projects whose behavior is highly context-dependent and will not work if other users try to take the same approach.

**Comment by tmat** [comment]
> will not work if other users try to take the same approach.

I don't understand what "other users" and "same approach" means.

**Comment by sharwell** [comment]
> I don't understand what do you mean by "other users" and "same approach".

If a user finds this:
https://github.com/dotnet/project-system/blob/403bdfa10be406240415908d53141cfc7400ede9/src/Templates/Microsoft.NetStandard.FSharp.ProjectTemplates/Microsoft.NetStandard.FSharp.ProjectTemplates.csproj#L4-L5

And tries to copy something similar into their own VS extension project, it will not work.

RepoToolset should be updated to use `DeployExtension` as the controlling property here, so projects with RepoToolset and projects without it have consistent behavior with respect to this property.

**Comment by tmat** [comment]
> And tries to copy something similar into their own VS extension project, it will not work.

Right. The toolset implements an abstraction over VS SDK since VS SDK is broken and insufficient in many ways. This is unfortunate but necessary. If we could move the fixes and redesign VS SDK we could make our projects example projects that other customers can copy. But not at this point.

**Comment by sharwell** [comment]
This issue only concerns one property. Is there an issue with this property preventing its use in RepoToolset?

**Comment by tmat** [comment]
Is there a usage of this property that's not a temporary workaround for an issue like https://github.com/dotnet/project-system/blob/403bdfa10be406240415908d53141cfc7400ede9/src/Templates/Microsoft.NetStandard.FSharp.ProjectTemplates/Microsoft.NetStandard.FSharp.ProjectTemplates.csproj#L4-L5 is?

Projects in our repos shouldn't be setting this property in general. All VSIXes should be deployed in local build and in integration test CI and not deployed on other CI legs.

**Comment by tmat** [comment]
In other words the toolset determines when to deploy and when not to deploy. The project has no say.
